akka-stream/src/main/scala/akka/stream/javadsl/Tcp.scala

```diff
@@ -323,4 +468,63 @@ class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
         .map(new IncomingConnection(_))
         .mapMaterializedValue(_.map(new ServerBinding(_))(ec).toJava))
 
+  /**
+   * Creates a [[Tcp.ServerBinding]] instance which represents a prospective TCP server binding on the given `endpoint`
+   * where all incoming and outgoing bytes are passed through TLS.
+   *
+   * @see [[Tcp.bind()]]
+   */
+  def bindWithTls(
+      interface: String,
+      port: Int,
+      createSSLEngine: Supplier[SSLEngine]): Source[IncomingConnection, CompletionStage[ServerBinding]] = {
+    Source.fromGraph(
+      delegate
+        .bindWithTls(interface, port, createSSLEngine = () => createSSLEngine.get())
+        .map(new IncomingConnection(_))
+        .mapMaterializedValue(_.map(new ServerBinding(_))(ec).toJava))
+  }
+
+  /**
+   * Creates a [[Tcp.ServerBinding]] instance which represents a prospective TCP server binding on the given `endpoint`
+   * where all incoming and outgoing bytes are passed through TLS.
+   *
+   * @see [[Tcp.bind()]]
+   */
+  def bindWithTls(
+      interface: String,
+      port: Int,
+      createSSLEngine: Supplier[SSLEngine],
+      backlog: Int,
+      options: JIterable[SocketOption],
+      idleTimeout: Optional[java.time.Duration],
+      verifySession: JFunction[SSLSession, Optional[Throwable]],
+      closing: TLSClosing): Source[IncomingConnection, CompletionStage[ServerBinding]] = {
+    // FIXME halfClose unused #26689
```

Comment:
In javadsl there was a `halfClose` parameter but it was not used. In scaladsl there is no parameter and it's always `true`. Is the scaladsl right, and it's ok to remove it from javadsl too?
Issue #26689

Reply:
No idea, let's leave it like this until someone finds a reason to dig in. Should be easy to evolve since there's no default parameters here anyway